examples/ExactlyOnceBasic/Program.cs

```diff
@@ -0,0 +1,680 @@
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.IO;
+
+
+namespace Confluent.Kafka.Examples.ExactlyOnceBasic
+{
+    /// <summary>
+    ///     Main class of this EOS producer example.
+    /// </summary>
+    public class Program
+    {
+
+        /// <summary>
+        ///     Logs provided <paramref name="message"/>.
+        /// </summary>
+        public static void Log(string message)
+        {
+            Console.Error.WriteLine($"{DateTimeOffset.Now.ToUnixTimeSeconds()}|EXAMPLE|{message}");
+        }
+
+        private static async Task CreateTopic(IAdminClient adminClient, string name, int partitions)
+        {
+
+            try
+            {
+                await adminClient.CreateTopicsAsync(new List<TopicSpecification>
+                {
+                    new TopicSpecification
+                    {
+                        Name = name,
+                        NumPartitions = partitions
+                    }
+                });
+                Log($"Created topic {name}");
+            }
+            catch (KafkaException)
+            {
+                // Continue if topic already exists
+                Log($"Topic {name} already exists");
+            }
+        }
+
+        /// <summary>
+        ///     Main method.
+        /// </summary>
+        public static async Task Main(string[] args)
+        {
+            CancellationTokenSource cancelled = new CancellationTokenSource();
+            CancellationTokenSource completed = new CancellationTokenSource();
+            Console.CancelKeyPress += (_, e) =>
+            {
+                e.Cancel = true; // prevent the process from terminating.
+                cancelled.Cancel();
+            };
+            AppDomain.CurrentDomain.ProcessExit += (_, e) =>
+            {
+                cancelled.Cancel();
+                completed.Token.WaitHandle.WaitOne();
+            };
+
+            if (args.Length != 2)
+            {
+                Console.WriteLine("Usage:");
+                Console.WriteLine("create_words <transactional-id>");
+                Console.WriteLine("reverse_words <transactional-id>");
+                completed.Cancel();
+                return;
+            }
+
+            var command = args[0];
+            var transactionalId = args[1];
+
+            IConfiguration configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("./appsettings.json")
+                .Build();
+
+            var producerConfig = configuration.GetSection("Producer").Get<ProducerConfig>().ThrowIfContainsNonUserConfigurable();
+            var consumerConfig = configuration.GetSection("Consumer").Get<ConsumerConfig>().ThrowIfContainsNonUserConfigurable();
+            var generalConfig = configuration.GetSection("General");
```

Comment:
It would be nicer to define a class `Config`, and bind to that (in the same way as ProducerConfig etc works above) rather than writing it out explicitly.